### PR TITLE
DROID-2688 Version history | Api

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt [hunk 1]
```diff
@@ -545,4 +545,29 @@ sealed class Command {
     data class ProcessCancel(
         val processId: Id
     ) : Command()
+
+    data class GetVersions(
```
Comment (uburoiubu) [inline core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt:549]: could we wrap all these new commands into `VersionHistory` class?

File: core-models/src/main/java/com/anytypeio/anytype/core_models/history/Version.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import com.anytypeio.anytype.core_models.Id
+import com.anytypeio.anytype.core_models.ObjectView
+import com.anytypeio.anytype.core_models.Payload
+
+data class Version(
+    val id: Id,
+    val previousIds: List<Id>,
+    val authorId: Id,
+    val authorName: String,
+    val time: Long,
```
Comment (uburoiubu) [inline core-models/src/main/java/com/anytypeio/anytype/core_models/history/Version.kt:13]: `timestamp: Long`

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt (hunk 1 above)
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt:549]: yes

File: domain/src/main/java/com/anytypeio/anytype/domain/history/DiffVersions.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            objectId = params.objectId,
+            spaceId = params.spaceId,
+            currentVersion = params.currentVersion,
+            previousVersion = params.previousVersion
+        )
+        return repo.diffVersions(command)
+    }
+
+    data class Params(
+        val ctx: Id,
```
Comment (uburoiubu) [inline domain/src/main/java/com/anytypeio/anytype/domain/history/DiffVersions.kt:28]: what's the difference between `ctx` and `objectId`?

**Review by uburoiubu — APPROVED** [review]

File: domain/src/main/java/com/anytypeio/anytype/domain/history/DiffVersions.kt (hunk 3 above)
Comment (konstantiniiv) [reply domain/src/main/java/com/anytypeio/anytype/domain/history/DiffVersions.kt:28]: no difference, I'll fix it

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt (hunk 1 above)
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/Command.kt]: done

File: core-models/src/main/java/com/anytypeio/anytype/core_models/history/Version.kt (hunk 2 above)
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/history/Version.kt]: done

File: domain/src/main/java/com/anytypeio/anytype/domain/history/DiffVersions.kt (hunk 3 above)
Comment (konstantiniiv) [reply domain/src/main/java/com/anytypeio/anytype/domain/history/DiffVersions.kt]: done

**Pull request merged**